pandas/tests/series/test_constructors.py

```diff
@@ -142,6 +142,24 @@ def test_constructor_list_like(self):
             result = Series(obj, index=[0, 1, 2])
             assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize('input_vals', [
+        ([1, 2]),
+        ([1.0, 2.0, np.nan]),
+        (['1', '2']),
+        (pd.date_range('1/1/2011', periods=2)),
+        (pd.date_range('1/1/2011', periods=2, tz='US/Eastern')),
+        (pd.Interval(left=0, right=5)),
+    ])
+    def test_constructor_list_str(self, input_vals):
+        # GH 16605
+        # Ensure that data elements are converted to strings when
```

Comment:
the original issue also covers DataFrame, I am not sure this fix will handle this. Can you add a test for that as well?

Reply:
Yeah, the original issue covers DataFrame but shows that it works as expected. For example, from the original issue:
```
In [1]: int_list = [1, 2, 3]
In [2]: s3 = pd.Series(int_list, dtype=str)

# Expect str but get int
In [3]: print(type(s3[0]))
Out[3]:
<class 'int'>

In [4]: f3 = pd.DataFrame(int_list, dtype=str)

# Expect str and get str
In [5]: print(type(f3.iloc[0, 0]))
Out[5]:
<class 'str'>
```

Reply:
are things covered for DataFrame? ping when ready.

Reply:
I couldn't find any tests that directly covered the issue so I added a similar test in ``tests/frame/test_dtypes.py``